pydm/display_module.py

```diff
@@ -1,40 +1,247 @@
+        loaded display subclass.
+
+    Returns
+    -------
+    pydm.Display
+    """
+    # Add the intelligence module directory to the python path, so that
+    # submodules can be loaded.
+    # Eventually, this should go away, and intelligence modules should behave
+    # as real python modules.
+    module_dir = os.path.dirname(os.path.abspath(pyfile))
+    sys.path.append(module_dir)
+    temp_name = str(uuid.uuid4())
+
+    # Now load the intelligence module.
+    module = imp.load_source(temp_name, pyfile)
+    if hasattr(module, 'intelclass'):
+        cls = module.intelclass
+        if not issubclass(cls, Display):
+            raise ValueError(
+                "Invalid class definition at file {}. {} does not inherit from Display. Nothing to open at this time.".format(
+                    pyfile, cls.__name__))
+    else:
+        classes = [obj for name, obj in inspect.getmembers(module) if
+                   inspect.isclass(obj) and issubclass(obj,
+                                                       Display) and obj != Display]
+        if len(classes) == 0:
+            raise ValueError(
+                "Invalid File Format. {} has no class inheriting from Display. Nothing to open at this time.".format(
+                    pyfile))
+        if len(classes) > 1:
+            warnings.warn(
+                "More than one Display class in file {}. The first occurence (in alphabetical order) will be opened: {}".format(
+                    pyfile, classes[0].__name__), RuntimeWarning, stacklevel=2)
+        cls = classes[0]
+
+    try:
+        # This only works in python 3 and up.
+        module_params = inspect.signature(cls).parameters
+    except AttributeError:
+        # Works in python 2, deprecated in 3.0 and up.
+        module_params = inspect.getargspec(cls.__init__).args
+
+    # Because older versions of Display may not have the args parameter or the macros parameter, we check
+    # to see if it does before trying to use them.
+    kwargs = {}
+    if 'args' in module_params:
+        kwargs['args'] = args
+    if 'macros' in module_params:
+        kwargs['macros'] = macros
+    instance = cls(**kwargs)
+    instance._loaded_file = pyfile
+    return instance
 
 
 class Display(QWidget):
     def __init__(self, parent=None, args=None, macros=None, ui_filename=None):
         super(Display, self).__init__(parent=parent)
         self.ui = None
         self._ui_filename = ui_filename
-        self.load_ui(parent=parent, macros=macros)
+        self._loaded_file = None
+        self._args = args
+        self._macros = macros
+        self._previous_display = None
+        self._next_display = None
+        if ui_filename or self.ui_filename():
+            self.load_ui(macros=macros)
+
+    def loaded_file(self):
+        return self._loaded_file
+
+    def previous_display(self):
+        return self._previous_display
+
+    def set_previous_display(self, display):
```

Comment:
Looks like a job for `@property/def previous_display...`

Reply:
Addressed via https://github.com/slaclab/pydm/pull/590/commits/d2ffa3ea76acc6a99c30c4d5b29f9f829111ce47